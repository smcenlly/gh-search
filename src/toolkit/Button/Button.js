import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

const ButtonWrapped = styled(({ color, ...props }) => <button {...props} />)`
    border: none;
    background: ${({ color }) => color};
    border-radius: 3px;
    color: white;
    padding: 5px 12px;
    font-size: 12px;
    cursor: pointer;

    :disabled {
        background: lightgray;
        cursor: not-allowed;
    }
`;

const Button = (props) => {
    return <ButtonWrapped {...props} />;
};

Button.COLOR = {
    PRIMARY: "rgb(62, 106, 170)",
    SECONDARY: "rgb(63, 120, 185)"
};

Button.propTypes = {
    color: PropTypes.oneOf(Object.values(Button.COLOR))
};

Button.defaultProps = {
    color: Button.COLOR.SECONDARY
};

export default Button;
