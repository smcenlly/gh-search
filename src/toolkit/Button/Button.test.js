import React from "react";
import { shallow } from "enzyme";
import Button from "./Button";

describe("Button", function() {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<Button />);
    });

    it("renders without crashing", () => {
        expect(wrapper).toExist();
    });

    it("matches snapshot", () => {
        expect(wrapper).toMatchSnapshot();
    });
});
