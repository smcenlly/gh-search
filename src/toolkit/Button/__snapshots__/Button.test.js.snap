// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `
<Styled(Component)
  color="rgb(63, 120, 185)"
/>
`;
